source/core/solid_base.cc

```diff
@@ -144,17 +148,46 @@ SolidBase<dim, spacedim>::setup_particles()
           }
       }
 
-  std::vector<BoundingBox<spacedim>> all_boxes;
-  all_boxes.reserve(fluid_tria->n_locally_owned_active_cells());
-  for (const auto cell : fluid_tria->active_cell_iterators())
-    if (cell->is_locally_owned())
-      all_boxes.emplace_back(cell->bounding_box());
-  const auto tree        = pack_rtree(all_boxes);
-  const auto local_boxes = extract_rtree_level(tree, 1);
+  // Boost RTREE bounding boxes
+  //  std::vector<BoundingBox<spacedim>> all_boxes;
+  //  all_boxes.reserve(fluid_tria->n_locally_owned_active_cells());
+  //  for (const auto cell : fluid_tria->active_cell_iterators())
+  //    if (cell->is_locally_owned())
+  //      all_boxes.emplace_back(cell->bounding_box());
+  //  const auto tree        = pack_rtree(all_boxes);
+  //  const auto local_boxes = extract_rtree_level(tree, 1);
+
+  //   global_fluid_bounding_boxes =
+  //    Utilities::MPI::all_gather(mpi_communicator, local_boxes);
+
+  // Naive global bounding boxes
+  // Obtaining global bounding boxes
 
   std::vector<std::vector<BoundingBox<spacedim>>> global_fluid_bounding_boxes;
-  global_fluid_bounding_boxes =
-    Utilities::MPI::all_gather(mpi_communicator, local_boxes);
+
+  if (auto tria =
```

Comment:
Could this if/else loop be documented? I think that this "if" condition checks for parallel computing, but as I understand it the "if" and the "else" block both use locally owned cells